bee-protocol/src/workers/packets/header.rs

```diff
@@ -0,0 +1,33 @@
+// Copyright 2020-2021 IOTA Stiftung
+// SPDX-License-Identifier: Apache-2.0
+
+//! Header of the type-length-value encoding.
+
+const HEADER_TYPE_SIZE: usize = 1;
+const HEADER_LENGTH_SIZE: usize = 2;
+pub(crate) const HEADER_SIZE: usize = HEADER_TYPE_SIZE + HEADER_LENGTH_SIZE;
+
+/// A header for the type-length-value encoding.
+#[derive(Clone, Debug, Eq, PartialEq)]
+pub(crate) struct HeaderPacket {
+    /// Type of the packet.
+    pub(crate) packet_type: u8,
+    /// Length of the packet.
+    pub(crate) packet_length: u16,
+}
+
+impl HeaderPacket {
+    // TODO impl try_from
+    pub(crate) fn from_bytes(bytes: &[u8]) -> Self {
+        Self {
+            packet_type: bytes[0],
+            // TODO propagate error
```

Comment:
same funny comment about doing `unsafe` magic.

Reply:
`unsafe` magic is not real. You can add a comment explaining that this will get optimized to not even matching the error because it can never happen.